contract_api/registry.py

```diff
@@ -0,0 +1,273 @@
+                filter_query, values = self._filters_to_query(
+                    qry_param.get("filters"))
+                print("get_all_srvcs::filter_query: ",
+                      filter_query, "|values: ", values)
+            total_count = self._get_total_count(
+                sub_qry=sub_qry, filter_query=filter_query, values=values)
+            if total_count == 0:
+                return self._search_response_format(total_count, offset, limit, [])
+            q_dta = self._srch_qry_dta(sub_qry=sub_qry, sort_by=sort_by, order_by=order_by,
+                                       offset=offset, limit=limit, filter_query=filter_query, values=values)
+            return self._search_response_format(total_count, offset, limit, q_dta)
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def get_group_info(self, org_id, service_id):
+        try:
+            group_data = self.repo.execute(
+                "SELECT G.*, E.* FROM service_group G INNER JOIN service_endpoint E ON G.group_id = E.group_id WHERE "
+                "G.org_id = %s AND G.service_id = %s ", [org_id, service_id])
+            self.obj_utils.clean(group_data)
+            groups = {}
+            for rec in group_data:
+                group_id = rec['group_id']
+                if group_id not in groups.keys():
+                    groups = {group_id: {"group_id": rec['group_id'],
+                                         "group_name": rec['group_name'],
+                                         "pricing": json.loads(rec['pricing']),
+                                         "endpoints": []
+                                         }
+                              }
+                    groups[group_id]['endpoints'].append({"endpoint": rec['endpoint'], "is_available":
+                                                          rec['is_available'], "last_check_timestamp": rec["last_check_timestamp"]})
+            return list(groups.values())
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _get_is_available_service(self):
+        try:
+            available_service = []
+            srvc_st_dta = self.repo.execute(
+                "SELECT DISTINCT org_id, service_id  FROM service_endpoint WHERE is_available = 1")
+            for rec in srvc_st_dta:
+                available_service.append((rec['org_id'], rec['service_id']), )
+            return available_service
+        except Exception as err:
+            print(repr(err))
+
+    def _filter_condition_to__query(self, filter_condition):
+        value = []
+        if filter_condition.attr in ["org_id", "service_id"]:
+            filter_condition.attr = "M." + filter_condition.attr
+        if filter_condition.operator == "=":
+            value = filter_condition.value
+            return '%s %s"%s"' % (filter_condition.attr, filter_condition.operator, "%s"), value
+        if filter_condition.operator == "IN":
```

Comment:
What happens if the value is a number? Do we end up placing single quotes around each value?